routers/api/v1/repo/issue_subscription.go

```diff
@@ -0,0 +1,221 @@
+	// ---
+	// summary: Unsubscribe user from issue
+	// consumes:
+	// - application/json
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: owner
+	//   in: path
+	//   description: owner of the repo
+	//   type: string
+	//   required: true
+	// - name: repo
+	//   in: path
+	//   description: name of the repo
+	//   type: string
+	//   required: true
+	// - name: index
+	//   in: path
+	//   description: index of the issue
+	//   type: integer
+	//   format: int64
+	//   required: true
+	// - name: user
+	//   in: path
+	//   description: user witch unsubscribe
+	//   type: string
+	//   required: true
+	// responses:
+	//   "201":
+	//     "$ref": "#/responses/empty"
+	//   "304":
+	//     description: User can only subscribe itself if he is no admin
+	//   "404":
+	//     description: Issue not found
+	issue, err := models.GetIssueByIndex(ctx.Repo.Repository.ID, ctx.ParamsInt64(":index"))
+	if err != nil {
+		if models.IsErrIssueNotExist(err) {
+			ctx.NotFound()
+		} else {
+			ctx.Error(500, "GetIssueByIndex", err)
+		}
+
+		return
+	}
+
+	user, err := models.GetUserByName(ctx.Params(":user"))
+	if err != nil {
+		if models.IsErrUserNotExist(err) {
+			ctx.NotFound()
+		} else {
+			ctx.Error(500, "GetUserByName", err)
+		}
+
+		return
+	}
+
+	//only admin and user for itself can change subscription
+	if user.ID != ctx.User.ID && !ctx.User.IsAdmin {
+		ctx.Error(403, "User", nil)
+		return
+	}
+
+	if err := models.CreateOrUpdateIssueWatch(user.ID, issue.ID, false); err != nil {
+		ctx.Error(500, "CreateOrUpdateIssueWatch", err)
+		return
+	}
+
+	ctx.Status(201)
+}
+
+// GetIssueSubscribers return subscribers of an issue
+func GetIssueSubscribers(ctx *context.APIContext, form api.User) {
+	// swagger:operation GET /repos/{owner}/{repo}/issues/{index}/subscriptions issue issueSubscriptions
+	// ---
+	// summary: Get users who subscribed on an issue.
+	// consumes:
+	// - application/json
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: owner
+	//   in: path
+	//   description: owner of the repo
+	//   type: string
+	//   required: true
+	// - name: repo
+	//   in: path
+	//   description: name of the repo
+	//   type: string
+	//   required: true
+	// - name: index
+	//   in: path
+	//   description: index of the issue
+	//   type: integer
+	//   format: int64
+	//   required: true
+	// responses:
+	//   "201":
+	//     "$ref": "#/responses/empty"
+	//   "404":
+	//     description: Issue not found
+	issue, err := models.GetIssueByIndex(ctx.Repo.Repository.ID, ctx.ParamsInt64(":index"))
+	if err != nil {
+		if models.IsErrIssueNotExist(err) {
+			ctx.NotFound()
+		} else {
+			ctx.Error(500, "GetIssueByIndex", err)
+		}
+
+		return
+	}
+
+	iw, err := models.GetIssueWatchers(issue.ID)
+	if err != nil {
+		ctx.Error(500, "GetIssueWatchers", err)
+		return
+	}
+
+	var subscribers []*api.User
+	for _, s := range iw {
+		if s.IsWatching {
+			user, err := models.GetUserByID(s.UserID)
```

Comment:
How about use a batch operation to get user information?

Reply:
@lunny added:
 * func (users UserList) APIFormat() []*api.User {}
 * func (iwl IssueWatchList) loadWatchUsers(e Engine) UserList {}
and changed DB query to batch